src/ICeWall/doc/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Data definitions
+
+ICeWall acts upon the information provided in the data files to produce a padring layout. The required data is split across 3 files to allow for the anticipated use models
+
+### Library data
+
+The library data is defined by calling the Footpring library function. This function takes a single argument, which is a TCL dictionary structure that defines the needed defintions of the cells.
+
+The following keys need to be defined in the dictionary
+* types
+* connect_by_abutment
+* pad_pin_name
+* pad_pin_layer
+* breakers
+* cells
+
+The types entry associates types used in the package data file with cells defined in the cells section, and in addition requires the defintion of the the corner cell to be used and the list of pad filler cells to be used ordered by cell width.
+
+The connect_by_abutment key defines the list of pins on the edges of the IO cells that are connected by abutment forming a ring of these signals around the die. ICeWall will connect these abuted signals together into SPECIAL netst, so that the router will not try to add wiring for these pins.
+
+The pad_pin_name key defines the name of the pin of the IO cell that is connected to the chip package.
+
+The pad_pin_layer key defines the layer that will be used to create a physical pin on the die that is coincident with the location of the external connection to the chip package
+
+The breakers key defines the list of cell types in the types definition that act as signal breakers for the signals that connect by abutment.
+
+The cells key defines additional data for each IO cell. Each type defined in the types key must have a matching definition in the list of cells. Within a cell definition, the following keys may be used
+
+* cell__name
+* orient
+* physical_only
+* breaks
+* signal_type 
+* power_pad
+* ground_pad
+* pad_pin_name
+
+| Key | Description |
+| cell_name | Can either be the name of the LEF macro to use, or else a dictionary allowing a different cell name to be specified for each side of the die. Sides are denoted by bottom right top and left |
+| orient | A dictionary that specifies the orientation of the cell for each side of the die |
+| physical_only | Cells that are not associated with cells in the design, primarily used for physical connections through the pad ring |
+| breaks | For breaker cells, defines the abutment signals which are broken by the presence of this cell. Each signal broken requires a list of pins on the breaker cell for the left and right hand side of the cell (An empty list may be provided if there are no pins for the broken signals |
+| signal_type | Set to 1 if this cell is to be regarded as a signal type |
+| power_pad | Optional. Set to 1 to define this cell as a power pad |
+| ground_pad | Optional. Set to 1 o define this cell as a ground pad |
+| pad_pin_name | Optional. Define the name of the pin on the macro that connects to the chip package |
```

Comment:
This table isn't getting rendered properly

Reply:
I think the table header is required so you want
| Key | Description |
**| ---| ---|**

Reply:
Done